src/java.base/share/classes/java/lang/String.java

```diff
@@ -930,66 +934,70 @@ private static byte[] encodeWithEncoder(Charset cs, byte coder, byte[] val, bool
                 cr.throwException();
         } catch (CharacterCodingException x) {
             if (!doReplace) {
-                throw new IllegalArgumentException(x);
+                @SuppressWarnings("unchecked")
+                E cce = (E) x;
+                throw cce;
             } else {
                 throw new Error(x);
             }
         }
         return trimArray(ba, bb.position());
     }
 
-    /*
-     * Throws iae, instead of replacing, if unmappable.
+    /**
+     * {@return the sequence of bytes obtained by encoding the given string in UTF-8}
+     *
+     * @param s the string to encode
+     * @throws NullPointerException If {@code s} is null
+     * @throws CharacterCodingException For malformed input or unmappable characters
      */
-    static byte[] getBytesUTF8NoRepl(String s) {
-        return encodeUTF8(s.coder(), s.value(), false);
+    static byte[] getBytesUTF8NoReplacement(String s) throws CharacterCodingException {
+        return encodeUTF8NoReplacement(s.coder(), s.value());  // Implicit null check on `s`
     }
 
     private static boolean isASCII(byte[] src) {
         return !StringCoding.hasNegatives(src, 0, src.length);
     }
 
-    /*
-     * Throws CCE, instead of replacing, if unmappable.
-     */
-    static byte[] getBytesNoRepl(String s, Charset cs) throws CharacterCodingException {
-        try {
-            return getBytesNoRepl1(s, cs);
-        } catch (IllegalArgumentException e) {
-            //getBytesNoRepl1 throws IAE with UnmappableCharacterException or CCE as the cause
-            Throwable cause = e.getCause();
-            if (cause instanceof UnmappableCharacterException) {
-                throw (UnmappableCharacterException)cause;
-            }
-            throw (CharacterCodingException)cause;
-        }
-    }
-
-    private static byte[] getBytesNoRepl1(String s, Charset cs) {
-        byte[] val = s.value();
+    /**
+     * {@return the sequence of bytes obtained by encoding the given string in
+     * the specified {@linkplain java.nio.charset.Charset charset}}
```

Comment:
Omit the link, for this private use, just say `Charset`.

Reply:
See my earlier comment – fixed in 01d4f878b4d.